Support for newer versions of zod

I'd like to use a newer version of zod but am running into peer dependency issues.

```
$ npm ci
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @asteasolutions/zod-to-openapi@1.0.1
npm ERR! Found: zod@3.17.3
npm ERR! node_modules/zod
npm ERR!   zod@"^3.16.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer zod@"~3.14.4" from @asteasolutions/zod-to-openapi@1.0.1
npm ERR! node_modules/@asteasolutions/zod-to-openapi
npm ERR!   @asteasolutions/zod-to-openapi@"^1.0.1" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: zod@3.14.5
npm ERR! node_modules/zod
npm ERR!   peer zod@"~3.14.4" from @asteasolutions/zod-to-openapi@1.0.1
npm ERR!   node_modules/@asteasolutions/zod-to-openapi
npm ERR!     @asteasolutions/zod-to-openapi@"^1.0.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /home/<redacted>/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/<redacted>/.npm/_logs/2022-06-28T20_05_29_528Z-debug-0.log
```

--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
     "openapi3-ts": "^2.0.2"
   },
   "peerDependencies": {
-    "zod": "^3.0.0"
+    "zod": "^3.14.0"
   },
   "devDependencies": {
     "@types/jest": "^27.4.1",
@@ -38,7 +38,7 @@
     "ts-jest": "^27.1.4",
     "typescript": "^4.6.3",
     "yaml": "^2.0.0",
-    "zod": "^3.0.0"
+    "zod": "^3.14.0"
   },
   "author": "Astea Solutions <info@asteasolutions.com>",
   "license": "MIT"

--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
     "openapi3-ts": "^2.0.2"
   },
   "peerDependencies": {
-    "zod": "~3.14.4"
+    "zod": "^3.0.0"
   },
   "devDependencies": {
     "@types/jest": "^27.4.1",
@@ -38,7 +38,7 @@
     "ts-jest": "^27.1.4",
     "typescript": "^4.6.3",
     "yaml": "^2.0.0",
-    "zod": "~3.14.4"
+    "zod": "^3.0.0"
   },
   "author": "Astea Solutions <info@asteasolutions.com>",
   "license": "MIT"

